### ISSUE TITLE
Longer transitions support

### ISSUE THREAD

**Issue by badstraw360** [sidoh/esp8266_milight_hub#513]
Is there any chance of making longer transitions, for example 10 minutes, instead of 1 minute only? (it doesn't work for more than 60 seconds) 

That would be great to make a Sunrise and sunset effect. 

Thanks again for this amazing software.

**Comment by poudenes** [comment]
You can also do this in your script. I had it in past with Domoticz dZvents

**Comment by sidoh** [comment]
There is not a limit on the length of transitions.  Duration is measured in seconds, and you should be able to pass any value (within reason).  Under the hood, this is converted to milliseconds.  The datatype being used is `size_t`, which is a 32-bit unsigned integer.  So the maximum transition time should be on the order of a month.  

What are you trying that isn't working?

**Comment by badstraw360** [comment]
Im using the service dev tool in HASS:

```
{
  "entity_id": "light.room",
  "transition": 600,
  "brightness_pct": 100
}

```

Instead of fading in 10 minutes, the transition lasts 1 minute. So If it's longer than 60 seconds it always transitions in 1 minute, but if it's less then it works as expected.

**Comment by sidoh** [comment]
Ah, you're definitely right.  Something is getting screwed up with the period calculation.

**Comment by sidoh** [comment]
Yeah, I see what's going on.  It's parameterizing on period, which is never specified when using homeassistant-controlled transitions.  So the default of 225ms is used, meaning the longest possible duration without manually specifying a period is roughly 100x that, or 22.5s.

I'll move things around to force a given duration.

**Comment by sidoh** [comment]
Hoping this commit does the trick: https://github.com/sidoh/esp8266_milight_hub/commit/8529d7ac96cf81052767b5d407c7f5dcba70998c

**Comment by sidoh** [comment]
Should be fixed in 1.10.1.  Thanks for reporting.
